File: Project 2 - Socket Programming/Python/UDPClient.py
# Project 2 - UDPCliet.py			
# Student Name: Joao Paulo D. S. Ferreira	
# CSIT 340-01 - Computer Networks 		
# Dr. Dawei Li		
# March 14, 2019				

# This program implements a UDP connection
# from the client side by using Socket Programming

from socket import *

# starts the connection with the server
serverHost = 'localhost'
serverPort = 5000
clientSocket = socket(AF_INET, SOCK_DGRAM)

message = input("Input a lowercase sentece: ")

# Continues sending requests until the user types Quit
while(message != "Quit"):
    
    # Sends the sentence typed by the user to the server
    # and prints the modified sentence recieve from the server
    clientSocket.sendto(message.encode(), (serverHost, serverPort))
    modifiedMessage, serverAddress = clientSocket.recvfrom(2048)
    print(modifiedMessage.decode())
    message = input("Input a lowercase sentece: ")

# closes the connection after the user types Quit 
clientSocket.close()

File: Project 2 - Socket Programming/Java/UDP_Client.java
/*
 * Project 2 - UDP_Client.java
 * Joao Paulo D. S. Ferreira
 * CSIT 340 - Computer Networks
 * Dr. Dawei Li
 * March 25, 2019
 */

/* 
 * This program contains the client side of a UDP connection 
 * It sends a sentence to the server and receives it back in upper case.
 */

import java.io.*;
import java.net.*;
import java.util.Scanner;

public class UDP_Client {
	
	private final static int PORT = 5000;
	private final static String HOSTNAME = "localhost";
	
	public static void main(String[] args) {
		System.out.println("This is the UDP Client.");
		
		Scanner scanner = new Scanner(System.in);
		System.out.print("Enter a sentence to convert to upper case. Type quit to quit: ");
		String requestString = scanner.nextLine();
		
		/* continue sending requests to the server until the user types quit */
		while(!requestString.toLowerCase().equalsIgnoreCase("quit")) {
			try {
				DatagramSocket socket = new DatagramSocket(0);
				

				/* sends the sentence to the server */
				byte[] requestBuffer = requestString.getBytes();
				InetAddress host = InetAddress.getByName(HOSTNAME);
				DatagramPacket request = new DatagramPacket(requestBuffer, requestBuffer.length, host, PORT);
				socket.send(request);
				
				/* receives the sentence back from the server */
				DatagramPacket response = new DatagramPacket(new byte[1024], 1024);
				socket.receive(response);
				String result = new String(response.getData());
				System.out.println(result);
				
				/* asks the user for another sentence */
				System.out.print("Enter a sentence to convert to upper case. Type quit to quit: ");
				requestString = scanner.nextLine();
				
			} catch (IOException e) {
				e.printStackTrace();
			}
		}
		scanner.close();
	}
}
